packages/client/lib/components/Map.jsx

```diff
@@ -17,35 +27,58 @@ const SetViewOnUserLocation = () => {
     })();
   }, []);
 
+  // for checking if map is mounted or not, used in getting current user location
+  useEffect(() => {
+    isMounted.current = true;
+
+    return () => {
+      isMounted.current = false;
+    };
+  }, []);
+
   useEffect(() => {
     navigator.geolocation.getCurrentPosition(
       (position) => {
-        map.setView(
-          { lat: position.coords.latitude, lng: position.coords.longitude },
-          12
-        );
+        if (!map) return;
+
+        if (isMounted.current) {
+          map.setView(
+            { lat: position.coords.latitude, lng: position.coords.longitude },
+            12
+          );
+          setLocation({
+            lat: position.coords.latitude,
+            lng: position.coords.longitude,
+          });
+        }
       },
       (error) => {
         console.error(error);
       },
       { enableHighAccuracy: true, timeout: 10000, maximumAge: Infinity }
     );
-  }, [map]);
+  }, [map, setLocation]);
 
   return null;
 };
 
 const Map = () => {
   const geoData = useStore((state) => state.geoData);
+  const markers = useStore((state) => state.markers);
 
   return (
-    <MapContainer center={geoData} zoom={12} zoomControl={false} className="min-h-screen w-full">
+    <MapContainer
+      center={geoData}
+      zoom={12}
+      zoomControl={false}
+      className="min-h-screen w-full"
+    >
       <SetViewOnUserLocation />
-      <TileLayer
-        attribution='&copy; <a href="http://osm.org/copyright">OpenStreetMap</a> contributors'
-        url="https://{s}.tile.openstreetmap.org/{z}/{x}/{y}.png"
-      />
-      <Marker position={geoData} draggable />
+      <TileLayer url="https://{s}.tile.openstreetmap.org/{z}/{x}/{y}.png" />
+      {markers.length > 0 &&
+        markers.map((marker, index) => {
+          return <Marker key={index} position={marker.location} />;
+        })}
```

Comment:
```suggestion
   markers.map((marker) =>  <Marker key={marker.id} position={marker.location} />)}
```
and we can add an `id` to the markers object from the `formSubmit`. Do you know if firestore will allow us to generate our own ids or we should get an id from firestore when we create a new form report? Just thinking ahead

Reply:
Firestore can auto generate ids for us.

Reply:
I refactored the code for now to generate an id for each marker to use Date.now() for now. We can refactor later if we want.